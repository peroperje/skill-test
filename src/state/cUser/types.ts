import { PayloadAction, TypeConstant } from 'typesafe-actions';
export enum CUserActionsTypes {
  LOGIN_REQUEST = '@cUser/LOGIN_REQUEST',
  LOGIN_SUCCESS = '@cUser/LOGIN_SUCCESS',
  LOGIN_FAILED = '@cUser/LOGIN_FAILED'
}

export interface CUser {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
}

export interface CUserState {
  data: CUser;
  fetching: boolean;
  error: string;
}

export interface PayloadRequest {
  email: string;
  password: string;
}

export type CUserRequestAction = PayloadAction<TypeConstant, PayloadRequest>;
export type CUserSuccessAction = PayloadAction<TypeConstant, CUser>;
export type CUserFailedAction = PayloadAction<TypeConstant, string>;
