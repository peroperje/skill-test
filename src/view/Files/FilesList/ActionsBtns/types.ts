export interface PropsId {
  id: number;
}

export interface PropsDownLoad {
  download: string;
}

export interface ActionsDipatch {
  action: () => void;
}
