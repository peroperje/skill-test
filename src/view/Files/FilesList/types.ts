import { FileItem } from '../../../state/files/types';

export interface FileListProps {
  files: FileItem[];
}
