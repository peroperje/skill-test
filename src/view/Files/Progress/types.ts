export interface StateProps {
  showProgress: boolean;
}
