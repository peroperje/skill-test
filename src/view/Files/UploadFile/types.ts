export interface SubmitDispatch {
  submit: (data: FileList) => void;
}
