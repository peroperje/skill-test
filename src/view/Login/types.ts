import { PayloadRequest } from '../../state/cUser/types';
export interface ActionsProp {
  submit: (data: PayloadRequest) => void;
}
