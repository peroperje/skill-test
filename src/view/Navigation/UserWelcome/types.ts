export interface StateProps {
  firstName: string;
  lastName: string;
}

export interface DispatchProps {
  logout: () => void;
}
