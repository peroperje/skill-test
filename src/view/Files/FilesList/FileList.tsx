import React from 'react';
import { FileListProps } from './types';
import ActionsBtns from './ActionsBtns/ActionsBtns';

const FilesList: React.FC<FileListProps> = ({
  files
}: FileListProps): JSX.Element => {
  const isFileArrEmpty = files.length === 0;
  return (
    <>
      {isFileArrEmpty && 'No files'}
      {!isFileArrEmpty && (
        <div className="Flex-column" style={{ width: '600px' }}>
          <div className="Flex-row">
            <div>File name</div>
            <div>Uploaded on</div>
            <div>Action</div>
          </div>
          {files.map(({ id, name, updateAt, download }) => {
            return (
              <div key={id} className="Flex-row">
                <div>{name}</div>
                <div>{updateAt}</div>
                <ActionsBtns id={id} download={download} />
              </div>
            );
          })}
        </div>
      )}
    </>
  );
};

export default FilesList;
