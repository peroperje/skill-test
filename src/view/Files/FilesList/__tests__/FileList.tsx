import React from 'react';
import { create } from 'react-test-renderer';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import { FileItem } from '../../../../state/files/types';
import FileList from '../FileList';

describe('FileList', () => {
  it('Snapshot empty files', () => {
    const component = create(<FileList files={[]} isFetching={false} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Shallow Snapshot with Files ', () => {
    const renderer = ShallowRenderer.createRenderer();
    const file: FileItem[] = [
      {
        id: 4,
        name: 'some name.csv',
        download: '45645--some-name.csv',
        updateAt: 45465464
      }
    ];
    const component = renderer.render(
      <FileList files={file} isFetching={false} />
    );
    expect(component).toMatchSnapshot();
  });
});
