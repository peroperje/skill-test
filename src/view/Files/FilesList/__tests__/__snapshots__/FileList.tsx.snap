// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList Shallow Snapshot with Files  1`] = `
<div
  className="makeStyles-tableWrapper-1"
>
  <ForwardRef(WithStyles)
    stickyHeader={true}
  >
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          align="center"
        >
          File Name
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          align="center"
        >
          Upload On
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          align="center"
        >
           
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          align="center"
        >
          some name.csv
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          align="center"
        >
          45465464
        </ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          align="center"
        >
          <ActionsBtns
            download="45645--some-name.csv"
            id={4}
          />
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</div>
`;

exports[`FileList Snapshot empty files 1`] = `
<div
  className="makeStyles-tableWrapper-1"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
  >
    There are no files
  </h4>
</div>
`;
